packages/grid/x-data-grid/src/utils/fastShallowCompare.ts

```diff
@@ -0,0 +1,34 @@
+const is = Object.is;
+
+export function fastShallowCompare<T extends Record<string, unknown> | null | undefined>(
```

Comment:
Indeed. It would return `true` for something like `[1, 2, 3]` and `{0: 1, 1: 2, 2: 3}` though.

I think we should not include arrays here, if the goal is "fast" then arrays can have an even faster specialized comparison function thanks to their `.length` prop and `while (i--)`.

Reply:
Done